File: MultiHotelManagement/src/main/java/com/MHM/MultiHotelManagement/service/BookingRoomService.java
package com.MHM.MultiHotelManagement.service;

<<<<<<< Updated upstream
import com.MHM.MultiHotelManagement.dto.request.BookingRoomRequestDTO;
import com.MHM.MultiHotelManagement.dto.response.BookingRoomResponseDTO;

import java.util.List;

public interface BookingRoomService {

    // BookingRoom তৈরি করো
    BookingRoomResponseDTO create(BookingRoomRequestDTO dto);

    // Booking এর জন্য multiple rooms একসাথে add করো
    List<BookingRoomResponseDTO> createMultiple(
            List<BookingRoomRequestDTO> dtoList
    );

    // ID দিয়ে খোঁজো
    BookingRoomResponseDTO getById(Long id);

    // Booking ID দিয়ে সব খোঁজো
    List<BookingRoomResponseDTO> getByBookingId(Long bookingId);

    // Room ID দিয়ে সব খোঁজো
    List<BookingRoomResponseDTO> getByRoomId(Long roomId);

    // Booking এর total price
    Double getTotalPriceByBookingId(Long bookingId);

    // Update করো
    BookingRoomResponseDTO update(
            Long id,
            BookingRoomRequestDTO dto
    );

    // একটা delete করো
    void delete(Long id);

    // Booking এর সব BookingRoom delete করো
    void deleteAllByBookingId(Long bookingId);
}
=======
public interface BookingRoomService {
}
>>>>>>> Stashed changes

File: S  A e/MultiHotelManagementAngular/src/app/services/booking-room.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { BookingRoom, BookingRoomRequest } from '../models/booking-room.model';
import { environment } from '../../environments/environments';

@Injectable({ providedIn: 'root' })
export class BookingRoomService {
  private API_URL = environment.apiUrl + 'booking-rooms';

  constructor(private http: HttpClient) {}

  getById(id: number) {
    return this.http.get<BookingRoom>(`${this.API_URL}/${id}`);
  }

  getByBookingId(bookingId: number) {
    return this.http.get<BookingRoom[]>(`${this.API_URL}/booking/${bookingId}`);
  }

  getByRoomId(roomId: number) {
    return this.http.get<BookingRoom[]>(`${this.API_URL}/room/${roomId}`);
  }

  getTotalPrice(bookingId: number) {
    return this.http.get<number>(`${this.API_URL}/booking/${bookingId}/total-price`);
  }

  create(dto: BookingRoomRequest) {
    return this.http.post<BookingRoom>(this.API_URL, dto);
  }

  createMultiple(dtoList: BookingRoomRequest[]) {
    return this.http.post<BookingRoom[]>(`${this.API_URL}/multiple`, dtoList);
  }

  update(id: number, dto: BookingRoomRequest) {
    return this.http.put<BookingRoom>(`${this.API_URL}/${id}`, dto);
  }

  delete(id: number) {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }

  deleteAllByBookingId(bookingId: number) {
    return this.http.delete<void>(`${this.API_URL}/booking/${bookingId}`);
  }
}
